ffi/src/scan.rs

```diff
@@ -398,5 +429,5 @@ pub unsafe extern "C" fn visit_scan_data(
         callback,
     };
     // TODO: return ExternResult to caller instead of panicking?
```

Comment:
reminder: get to this TODO!

Reply:
@OussamaSaoudi 

This isn't actually quite as easy at it seems. To become an `ExternResult` you need an `engine`, and as the APIs are now you likely won't have an engine when calling this, because `kernel_scan_data_next` doesn't have one, so it can't pass it in when it calls this function.

We could take an `engine` arg in `kernel_scan_data_next` and thread it all the way through, but that's a bigger change. So I'm making an issue and punting for now :) #680